tedana/metrics/external_regressor_fits.py

```diff
@@ -0,0 +1,751 @@
+        The R2 statistics for the fits to the full and partial models
+    """
+    betas_base, sse_base, df_base = fit_model(regressor_models[base_label], y)
+    betas_full, sse_full, df_full = fit_model(regressor_models["full"], y)
+
+    # larger sample variance / smaller sample variance (F = (SSE1 – SSE2 / m) / SSE2 / n-k,
+    # where SSE = residual sum of squares, m = number of restrictions and k = number of
+    # independent variables) -> the 'm' restrictions in this case is the DOF range between
+    # the base - full model, the 'n-k' is the number of DOF (independent variables/timepoints)
+    # from the fully-fitted model
+    f_vals = np.divide((sse_base - sse_full) / (df_base - df_full), (sse_full / df_full))
+    # cumulative distribution (FX(x) = P(X<=x), X = real-valued variable, x=probable variable
+    # within distribution) of the F-values + extra parameters to shape the range of the
+    # distribution values (range: start = DOF_base (unfitted) -
+    #       DOF_Full (fitted with full model), end = DOF_Full)
+    p_vals = 1 - stats.f.cdf(f_vals, df_base - df_full, df_full)
+    # estimate proportion of variance (R2-squared fit) by SSE full [fully fitted]
+    # (sum of 1st errors) / SSE base [non-fitted] (sum of 2nd errors) ... and subtracting
+    # the error ratio from 1: R² = SSR (fitted model)/SST(total or model base) =
+    # Σ (Y_pred-Y_actual)**2 / Σ (Y_pred-Y_actual)**2
+    r2_vals = 1 - np.divide(sse_full, sse_base)
+    print(y.shape)
+
+    # Vestigial code for testing that might be worth reviving
+    # Plots the fits for the first 20 components
+    # if show_plot:
+    #     plt.clf()
+    #     fig = plt.figure(figsize=(20, 24))
+    #     for idx in range(30):
+    #         # print('Outer bound index: ', idx)
+
+    #         if idx < Y.shape[1]:  # num of components
+    #             # print('Actual axis index: ', idx)
+    #             ax = fig.add_subplot(5, 6, idx + 1)  # this axis index starts from 1
+    #             plot_fit(
+    #                 ax,
+    #                 Y[:, idx],
+    #                 betas_full[:, idx],
+    #                 regressor_models["full"],
+    #                 betas_base=betas_base[:, idx],
+    #                 X_base=regressor_models[base_label],
+    #                 F_val=F_vals[idx],
+    #                 p_val=p_vals[idx],
+    #                 R2_val=R2_vals[idx],
+    #                 SSE_base=SSE_base[idx],
+    #                 SSE_full=SSE_full[idx],
+    #                 base_legend=base_label,
+    #             )
+    #     base_save_label = base_label.replace(" ", "_")
+    #     plt.savefig(
+    #         f"{prefix}_ModelFits_{base_save_label}.jpeg",
+    #         pil_kwargs={"quality": 20},
+    #         dpi="figure",
+    #     )  # could also be saved as eps
+
+    return betas_full, f_vals, p_vals, r2_vals
+
+
+def correlate_regressors(comptable, external_regressors, mixing, detrend_regressors=None):
```

Comment:
Given that F-stats are better than correlations, I propose we drop this entirely. I'd really like to avoid adding unnecessary flexibility.

Reply:
Done